packages/context/src/__tests__/unit/invocation-context.unit.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Context, InvocationContext} from '../..';
+
+describe('InvocationContext', () => {
+  let ctx: Context;
+  let invocationCtxForGreet: InvocationContext;
+  let invocationCtxForCheckName: InvocationContext;
+  let invalidInvocationCtx: InvocationContext;
+
+  before(givenContext);
+  before(givenInvocationContext);
+
+  it('has a getter for targetClass', () => {
+    expect(invocationCtxForGreet.targetClass).to.equal(MyController);
+    expect(invocationCtxForCheckName.targetClass).to.equal(MyController);
+  });
+
+  it('has a getter for targetName', () => {
+    expect(invocationCtxForGreet.targetName).to.equal(
+      'MyController.prototype.greet',
+    );
+    expect(invocationCtxForCheckName.targetName).to.equal(
+      'MyController.checkName',
+    );
+  });
+
+  it('has a getter for description', () => {
+    expect(invocationCtxForGreet.description).to.equal(
+      `InvocationContext(${
+        invocationCtxForGreet.name
+      }): MyController.prototype.greet`,
+    );
+    expect(invocationCtxForCheckName.description).to.equal(
+      `InvocationContext(${
+        invocationCtxForCheckName.name
+      }): MyController.checkName`,
+    );
+  });
+
+  it('throws error if method does not exist', () => {
+    expect(() => invalidInvocationCtx.assertMethodExists()).to.throw(
+      'Method MyController.prototype.invalid-method not found',
+    );
+  });
```

Comment:
nitpick: maybe test the error catch for static method as well.

Reply:
added